### DISCUSSION TITLE
Question about validity of storing $data proxy outside of Alpine.

### DISCUSSION THREAD

**Discussion by kylethielk** [alpinejs/alpine#921] [1. Help]
Hi Alpine community!

The following works as expected but I'm looking to get some clarification on the long-term viability of doing this (i.e will this be removed or made backwards incompatible in V3) and to see if there are any potential pitfalls I'm not foreseeing? Essentially I'm storing a reference to the Reactive Proxy outside the scope of Alpine and modifying the proxy directly.

```javascript
window.Container = {
    data: { person: {name: "John Doe"} }
}
//...called from other code, later in script life.
window.Container.data.person.name = "Jane Doe"; //UI Is properly updated reactively.
```

```html
<div x-data="Container.data" x-init="Container.data = $data">
        <input type="text" x-model="person.name" />
        <div x-text="person.name"></div>
</div>
```

Thanks for any feedback.

**Comment by KevinBatdorf** [comment]
Stylistically it wouldn't be my personal way of doing it. Maybe this reads better for you?

```js
document.querySelector('#mycomponent').__x.$data.person.name = "Jane Doe"
```

**Reply by kylethielk** [reply]
Thanks for reply @KevinBatdorf. I'm less concerned about style here, more concerned about using what appears to be an undocumented feature and use case (which I believe `__x` is as well).

As an aside, one of the reasons for us using a library like Alpine is to move away from our javascript being concerned with the low level DOM and having our UI updates being more data driven. In my experience tight coupling between our javascript and low level DOM has led to more brittle code in the past. For example a designer changing a class name or DOM tree can lead to broken selectors on the javascript side. What I love about Alpine is the super explicit DOM attributes (at the expense of brevity) that make it incredibly clear and obvious the interplay between HTML and data. All that to say, we are actively trying to move away from hard-coding query selectors like your example.

Thanks!

**Reply by KevinBatdorf** [reply]
I'm just a community member here but I'd bet on it not changing without a major version update, so you should be fine using either approach. You could also just lock to the current minor version of Alpine too if you want to be doubly careful about it. 

I agree completely. Alpine is great exactly for that reason.
